powertools-cloudformation/src/main/java/software/amazon/lambda/powertools/cloudformation/CloudFormationResponse.java

```diff
@@ -0,0 +1,249 @@
+
+    /**
+     * Indicates if the function invoking send had successfully completed.
+     */
+    public enum ResponseStatus {
+        SUCCESS, FAILED
+    }
+
+    /**
+     * Representation of the payload to be sent to the event target URL.
+     */
+    @SuppressWarnings("unused")
+    static class ResponseBody {
+        private final String status;
+        private final String reason;
+        private final String physicalResourceId;
+        private final String stackId;
+        private final String requestId;
+        private final String logicalResourceId;
+        private final boolean noEcho;
+        private final Object data;
+
+        ResponseBody(CloudFormationCustomResourceEvent event,
+                     Context context,
+                     ResponseStatus responseStatus,
+                     Object responseData,
+                     String physicalResourceId,
+                     boolean noEcho) {
+            requireNonNull(event, "CloudFormationCustomResourceEvent");
+            requireNonNull(context, "Context");
+            this.physicalResourceId = physicalResourceId != null ? physicalResourceId : context.getLogStreamName();
+            this.reason = "See the details in CloudWatch Log Stream: " + context.getLogStreamName();
+            this.status = responseStatus == null ? ResponseStatus.SUCCESS.name() : responseStatus.name();
+            this.stackId = event.getStackId();
+            this.requestId = event.getRequestId();
+            this.logicalResourceId = event.getLogicalResourceId();
+            this.noEcho = noEcho;
+            this.data = responseData;
+        }
+
+        public String getStatus() {
+            return status;
+        }
+
+        public String getReason() {
+            return reason;
+        }
+
+        public String getPhysicalResourceId() {
+            return physicalResourceId;
+        }
+
+        public String getStackId() {
+            return stackId;
+        }
+
+        public String getRequestId() {
+            return requestId;
+        }
+
+        public String getLogicalResourceId() {
+            return logicalResourceId;
+        }
+
+        public boolean isNoEcho() {
+            return noEcho;
+        }
+
+        public Object getData() {
+            return data;
+        }
+    }
+
+    /*
+     * Replace with java.util.Objects::requireNonNull when dropping JDK 8 support.
+     */
+    private static <T> T requireNonNull(T arg, String argName) {
+        if (arg == null) {
+            throw new NullPointerException(argName + " cannot be null.");
+        }
+        return arg;
+    }
+
+    private final SdkHttpClient client;
+    private final ObjectMapper mapper;
+
+    /**
+     * Creates a new CloudFormationResponse that uses the provided HTTP client and default JSON serialization format.
+     *
+     * @param client HTTP client to use for sending requests; cannot be null
+     */
+    public CloudFormationResponse(SdkHttpClient client) {
```

Comment:
Will be good to provide an implementation where user dont need to care about passing http client as well and module can take care of setting it up.

From Custom resource perspective, I believe it will provide better developer experience.

Reply:
True. The JS and Python versions embed the client. We'd just need to 
1. Pick a particular client impl (Apache?) and 
2. Include it as a dependency to the project.